jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/BuildService.java

```diff
@@ -31,11 +32,20 @@
     boolean isApplicable();
 
     /**
-     * Builds the given image using the specified configuration.
+     * Builds the given images using the specified configuration.
      *
-     * @param imageConfig the image to build
+     * @param imageConfigs the image to build
+     * @throws JKubeServiceException in case of any error while building image
      */
-    void build(ImageConfiguration imageConfig) throws JKubeServiceException;
+    void build(Collection<ImageConfiguration> imageConfigs) throws JKubeServiceException;
+
+    /**
+     * Build the given image using specified configuration
+     *
+     * @param imageConfiguration single {@link ImageConfiguration} to build
+     * @throws JKubeServiceException in case of any error while building image
+     */
+    void build(ImageConfiguration imageConfiguration) throws JKubeServiceException;
```

Comment:
Why did we keep the legacy method?

Reply:
I've shifted image build skip logic to BuildService interface. this method gets called when image build is not skipped

Reply:
After your update, I still see that we keep the two methods List and single image.

Is there a specific reason to keep both (at least in the public interface)??

Reply:
https://github.com/rohanKanojia/jkube/pull/60